extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java

```diff
@@ -697,9 +691,11 @@ private static void writeKeyValuePartitionedTableImpl(@NotNull final Partitioned
         }
         final Map<String, Map<ParquetCacheTags, Object>> computedCache =
                 buildComputedCache(() -> sourceTable.orElseGet(partitionedTable::merge), leafDefinition);
-        writeParquetTablesImpl(partitionedData.toArray(Table[]::new), leafDefinition, writeInstructions,
-                destinations.toArray(File[]::new), leafDefinition.getGroupingColumnNamesArray(),
-                partitioningColumnsSchema, new File(destinationRoot), computedCache);
+        final Table[] partitionedDataArray = partitionedData.toArray(Table[]::new);
+        // TODO Verify correctness with Ryan/Larry
```

Comment:
Work with Larry/Ryan

Reply:
In the branch Larry and I shipped, which only looks at single-table indexes, we've taken the second way.

You could do that, by assuming that:
1. A `PartitionedTable` will have indexes for you to write (or not) on its constituents. Conceivably, those constituents should be considered individually for writing indexes.
2. A Table that you're going to partition before writing will have indexes you can intersect. I wouldn't want the generic `partitionBy` to intersect parent table indexes, but we could easily do a `transform` to allow us to intersect the index tables' row sets with the constituent row sets, then filter the empty row sets.

Pseudocode for intersecting, probably with bad parentheses, etc:
```
partitionedSource = sourceTable.partitionBy(partitioningColumns)
for dataIndex in sourceTable indexes:
    indexTable = dataIndex.table()
    partitionedIndexTable = partitionedSource.transform(c ->
        // Do this with `FunctionalColumns` to avoid compiles
        indexTable
            .update(List.of(new FunctionalColumn(dataIndex.rowSetColumnName(), RowSet.class, dataIndex.rowSetColumnName(), RowSet.class, dataIndexRowSet -> dataIndexRowSet.intersect(c.getRowSet()))))
            .updateView(List.of(new FunctionalColumn(dataIndex.rowSetColumnName(), RowSet.class, "__NON_EMPTY__", Boolean.class, RowSet::isNonempty)))
            .where(Filter.isTrue(ColumnName.of("__NON_EMPTY__"))))
    // We *could* add `StandaloneDataIndex`es to the constituents, or just write from the partitioned index table
```

All of that said, I'm not opposed to using instructions, but so far we've just said "the user tells us what to index using the indexing APIs". If we're going to use instructions, we need to add that support to non-partitioned table writing code.

Reply:
Created issue #5292 for this